### PR TITLE
[Improvement](jdbc external table)Support jdbc external table for nereids.

### PR DESCRIPTION
**Pull request by Jibing-Li**
## Proposed changes

Issue Number: close #xxx

Nereids planner only support JDBC external catalog, this pr is to support JDBC external table for nereids.

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by morningman** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalJdbcScan.java [hunk 1]
```diff
@@ -31,28 +33,28 @@
...
+    public LogicalJdbcScan(ObjectId id, TableIf table, List<String> qualifier) {
         this(id, table, qualifier, Optional.empty(), Optional.empty());
     }
 
     @Override
-    public ExternalTable getTable() {
-        Preconditions.checkArgument(table instanceof ExternalTable);
-        return (ExternalTable) table;
+    public TableIf getTable() {
+        Preconditions.checkArgument(table instanceof ExternalTable || table instanceof JdbcTable);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalJdbcScan.java:56]: please add check error msg

**Comment by Jibing-Li** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**